apps/aecore/lib/aecore/structures/oracle_registration_tx_data.ex

```diff
@@ -0,0 +1,190 @@
+defmodule Aecore.Structures.OracleRegistrationTxData do
+  alias __MODULE__
+  alias Aecore.Structures.Account
+  alias Aecore.Wallet.Worker, as: Wallet
+  alias Aecore.Oracle.Oracle
+  alias Aecore.Chain.ChainState
+
+  require Logger
+
+  @type tx_type_state :: ChainState.oracles()
+
+  @type payload :: %{
+          query_format: map(),
+          response_format: map(),
+          query_fee: non_neg_integer(),
+          ttl: Oracle.ttl()
+        }
+
+  @type t :: %OracleRegistrationTxData{
+          query_format: map(),
+          response_format: map(),
+          query_fee: non_neg_integer(),
+          ttl: Oracle.ttl()
+        }
+
+  defstruct [
+    :query_format,
+    :response_format,
+    :query_fee,
+    :ttl
+  ]
+
+  @spec get_chain_state_name() :: :oracles
+  def get_chain_state_name(), do: :oracles
+
+  use ExConstructor
+
+  @spec init(payload()) :: OracleRegistrationTxData.t()
+  def init(%{
+        query_format: query_format,
+        response_format: response_format,
+        query_fee: query_fee,
+        ttl: ttl
+      }) do
+    %OracleRegistrationTxData{
+      query_format: query_format,
+      response_format: response_format,
+      query_fee: query_fee,
+      ttl: ttl
+    }
+  end
+
+  @spec is_valid?(OracleRegistrationTxData.t()) :: boolean()
+  def is_valid?(%OracleRegistrationTxData{
+        query_format: query_format,
+        response_format: response_format,
+        ttl: ttl
+      }) do
+    formats_valid =
+      try do
+        ExJsonSchema.Schema.resolve(query_format)
+        ExJsonSchema.Schema.resolve(response_format)
+        true
+      rescue
+        e ->
```

Comment:
why rescue instead of catch? use full error name, instead of e

Reply:
The resolve function raises an error instead of throwing, I also don't know what the possible errors are and they wouldn't be handled in a different way